db/data_migration/20170818104227_convert_unknown_news_articles_to_valid_news_articles.rb

```diff
@@ -0,0 +1,38 @@
+begin
+  unknown_news_articles = NewsArticle.by_subtype(NewsArticleType::Unknown)
+
+  # Update NewsArticles in Whitehall
+  press_releases = unknown_news_articles.includes(:document).select { |n| n.slug.include?("press-release") }
+  news_stories = unknown_news_articles - press_releases
+
+  press_releases.each do |news_article|
+    news_article.type = NewsArticleType::PressRelease
+
+    if !news_article.minor_change && news_article.change_note.nil?
+      news_article.change_note = "This NewsArticle was converted to a Press Release"
+    end
+
+    news_article.save!
+  end
+
+  news_stories.each do |news_article|
+    news_article.type = NewsArticleType::NewsStory
+
+    if !news_article.minor_change && news_article.change_note.nil?
+      news_article.change_note = "This NewsArticle was converted to a News Story"
+    end
+
+    news_article.save!
+  end
+
+  # Republish updated NewsArticles to PublishingAPI
+  document_ids = unknown_news_articles.map { |n| n.document.id }
```

Comment:
You should probably `uniq` these as there will be duplicates. Also if you use `n.document_id` it'll might loading the `Document`.